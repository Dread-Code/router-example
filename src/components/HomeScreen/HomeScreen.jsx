import React, { useContext } from "react";
import { UserContext } from "../../context/UserContext";

export const HomeScreen = () => {
  const { user } = useContext(UserContext);
  return (
    <div>
      <h1>HomeScreen</h1>
      <pre>{JSON.stringify(user)}</pre>
    </div>
  );
};
