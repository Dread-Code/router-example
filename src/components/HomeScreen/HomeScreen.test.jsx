import React from "react";
import { mount } from "enzyme";
import { HomeScreen } from "./HomeScreen";
import { UserContext } from "../../context/UserContext";

describe("HomeScreen", () => {
  const user = {
    name: "Lucas",
    email: "lucas@gmail.com",
  };
  const wrapper = mount(
    <UserContext.Provider
      value={{
        user,
      }}
    >
      <HomeScreen />
    </UserContext.Provider>
  );
  test("should match snapshoot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
