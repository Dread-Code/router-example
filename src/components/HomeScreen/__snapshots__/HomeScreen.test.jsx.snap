// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen should match snapshoot 1`] = `
<div>
  <h1>
    HomeScreen
  </h1>
  <pre>
    {"name":"Lucas","email":"lucas@gmail.com"}
  </pre>
</div>
`;
